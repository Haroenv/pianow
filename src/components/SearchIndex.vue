<template lang="pug">
  .search-index
    slot
      h1 You can now put a search-results or/and a search-box
</template>

<script>
import algoliasearchHelper from "algoliasearch-helper";

export default {
  name: "SearchIndex",
  inject: ["_client"],
  props: {
    name: {
      type: String,
      required: true
    },
    options: {
      type: Object,
      default: null
    }
  },
  data() {
    return {
      localHelper: algoliasearchHelper(this._client, this.name, Object.assign({}, this.options))
    };
  },
  mounted() {
    // trigger initial search
    this.localHelper.search();
  },
  provide() {
    return {
      _index: this.name,
      _helper: this.localHelper
    };
  }
};
</script>
