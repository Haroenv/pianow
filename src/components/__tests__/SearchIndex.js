import Vue from "vue";
import SearchIndex from "../SearchIndex";
import algoliasearchHelper from "algoliasearch-helper";

jest.mock("algoliasearch-helper");
algoliasearchHelper.mockImplementation(() => ({
  search: jest.fn()
}))

describe("SearchIndex.vue", () => {
  let vm;
  const client = "client"
  const options = {
    hasProperty: true
  };

  beforeEach(() => {
    algoliasearchHelper.mockClear();
  });

  it("has a mounted hook", () => {
    expect(typeof SearchIndex.mounted).toBe("function");
  });

  it("has a provide function which return an object with a _index and _helper properties", () => {
    expect(typeof SearchIndex.provide).toBe("function");
    const provide = SearchIndex.provide();
    expect(typeof provide).toBe("object");
    expect(provide.hasOwnProperty("_index")).toBeTruthy();
    expect(provide.hasOwnProperty("_helper")).toBeTruthy();
  });
  // function for creating Vue vm
  beforeEach(() => {
    vm = new Vue({
      template: `<search-index name="testName" :options="options"/>`,
      provide: {
        _client: client
      },
      data() {
        return {
          options
        };
      },
      components: {
        SearchIndex
      }
    }).$mount();
  });
  
  it("should create algoliasearch helper with correct component props", () => {
    expect(algoliasearchHelper).toHaveBeenCalledTimes(1);
    expect(algoliasearchHelper).toBeCalledWith(client, "testName", options);
  });

  it("should render correct contents", () => {
    expect(vm.$el.querySelector(".search-index h1").textContent).toEqual(
      "You can now put a search-results or/and a search-box"
    );
  });

  it("should match the snapshot", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
