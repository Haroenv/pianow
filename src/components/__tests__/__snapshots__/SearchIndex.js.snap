// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchIndex.vue should match the snapshot 1`] = `
<div
  class="search-index"
>
  <h1>
    You can now put a search-results or/and a search-box
  </h1>
</div>
`;
